File: mw-kreator-regul-biznesowych/app-modul-model/src/main/java/model/encje/Encja.java
package model.encje;

import javax.persistence.*;

@MappedSuperclass
public class Encja {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Version
    @Column(name = "WERSJA", columnDefinition = "integer DEFAULT 1", nullable = false)
    private Long wersja =1L ;


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getWersja() {
        return wersja;
    }

    public void setWersja(Long wersja) {
        this.wersja = wersja;
    }
}

File: mw-kreator-regul-biznesowych/app-modul-model/src/main/kotlin/model/encje/Encja.kt
package model.encje

import org.springframework.data.util.ProxyUtils
import javax.persistence.*

@MappedSuperclass
open class Encja{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @Version
    @Column(name = "WERSJA",columnDefinition = "integer DEFAULT 1", nullable = false)
    var wersja: Long=1


    override fun equals(other: Any?): Boolean {
        other ?: return false

        if (this === other) return true

        if (javaClass != ProxyUtils.getUserClass(other)) return false

        other as Encja

        return if (null == this.id) false else this.id == other.id
    }

    override fun hashCode(): Int {
        return 31
    }

    override fun toString() = "Entity of type ${this.javaClass.name} with id: $id"
}